File: admin-fe/src/app/service/khach-hang.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";

import { PagedResponse } from "../model/interface/paged-response.interface";
import { KhachHang } from "../model/class/khach-hang.class";

@Injectable({
  providedIn: "root",
})
export class KhachHangService {
  private readonly apiUrl = "http://localhost:8080/khach-hang";

  constructor(private http: HttpClient) {}

  public getAll(
    pageNumber: number = 1,
    pageSize: number = 5,
    search: string = ""
  ): Observable<PagedResponse<KhachHang>> {
    const param = `?pageNumber=${pageNumber}&pageSize=${pageSize}&search=${search}`;
    return this.http.get<PagedResponse<KhachHang>>(
      `${this.apiUrl}/get-all${param}`
    );
  }

  public add(khachHang: KhachHang): Observable<KhachHang> {
    return this.http.post<KhachHang>(`${this.apiUrl}/add-kh`, khachHang);
  }

  public getById(id: number): Observable<KhachHang> {
    return this.http.get<KhachHang>(`${this.apiUrl}/getById/${id}`);
  }

  public changeStatus(id: number): Observable<string> {
    return this.http.get(`${this.apiUrl}/delete/${id}`, {
      responseType: "text",
    });
  }

  public update(khachHang: KhachHang): Observable<KhachHang> {
    return this.http.put<KhachHang>(`${this.apiUrl}/update-kh`, khachHang);
  }
}

File: back-end/back-end/src/main/java/com/datn/backend/service/KhachHangService.java
package com.datn.backend.service;

import com.datn.backend.dto.request.KhachHangRequest;
import com.datn.backend.dto.response.KhachHangResponse;
import com.datn.backend.dto.response.NhanVienResponse;
import com.datn.backend.dto.response.PagedResponse;
import com.datn.backend.model.khach_hang.KhachHang;

import java.util.List;

public interface KhachHangService {
    KhachHang add(KhachHangRequest kh);
    PagedResponse<KhachHangResponse> getAll(int pageNumber, int pageSize, String search);
    KhachHang update(int id, KhachHangRequest kh);
    KhachHang delete(Integer id);
    KhachHangResponse getById(int id);
    PagedResponse<KhachHangResponse> filter(int pageNumber, int pageSize, List<Integer> gioiTinhFilter, List<Integer> trangThaiFilter);


}
